### DISCUSSION TITLE
How to modify the program if the SPI of the non-multiplexed screen is touched.

### DISCUSSION THREAD

**Discussion by fmzhangpei** [Bodmer/TFT_eSPI#1917] [Compatible displays and setup files]
How to modify the program if the SPI of the non-multiplexed screen is touched.

**Comment by Bodmer** [comment]
The library assumes the touch and display are on the same bus.

To use touch on anothr SPI bus you will need to use an additional thrid party touch library.

**Reply by glenhay** [reply]
Hello Bodmer, I recently purchased a display integrated with an ESP32 from Ali Express [here](https://www.aliexpress.com/item/1005004632953455.html?spm=a2g0o.productlist.0.0.76f53aa1JtiQ7M&algo_pvid=ffcdabfd-97de-4717-af8f-2bb80a383e39&aem_p4p_detail=2022092303131116880900759335470000203256&algo_exp_id=ffcdabfd-97de-4717-af8f-2bb80a383e39-4&pdp_ext_f=%7B%22sku_id%22%3A%2212000029911293172%22%7D&pdp_npi=2%40dis%21AUD%2123.25%2123.25%21%21%217.24%21%21%402101e9d216639279918164219e713c%2112000029911293172%21sea&curPageLogUid=2gw2r0KITPgy&ad_pvid=2022092303131116880900759335470000203256_1) and the documentation provided used some examples with your TFT_eSPI library. While I was initially impressed with the library I wasted a day trying to make the touch screen respond. After analysing some of the other examples that used the lvgl.h library I discovered that the display and touch screen are on different SPI buses and much searching lead me here. As these connections are hardwired it seems as though I am unable to use your library in its current form, is there any plan to define the TOUCH connections to other ios? Thanks for your good work

**Reply by Bodmer** [reply]
As noted above you can use third party touch libraries. TFT_eTouch is good and includes TFT_eSPI examples with separate SPI bus for the touch controller.

**Comment by fmzhangpei** [comment]
First of all, thank you very much for taking the time out of your busy schedule to answer my question. I am already working on adding a third-party library to this touch, and it should be done soon. The TFT_eSPI library, after your hard work, is indeed very powerful, and it is very suitable for the development of LCD screen interface for projects with a certain foundation, but I have a bigger problem now. I tried to compare the brushes developed by your library. The screen speed is compared with some domestic serial screens in China. The Chinese serial screen uses the STM32G031 chip, and the main frequency should be 64M. The same 4-wire SPI screen, the Chinese serial screen will be faster. Based on your library, Increasing the SPI main frequency will increase the speed and make me satisfied, but there is a problem that the main frequency of the screen driver chip, the highest frequency written in the specification is 10M, so if we use the SPI main frequency above 10M, it is in Overclocking, as a design product, lacks rigor and is risky. Is it possible to improve the screen refresh speed by other methods without making the SPI main frequency so high?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Bodmer/TFT_eSPI"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年7月21日(星期四) 晚上8:13
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Bodmer/TFT_eSPI] How to modify the program if the SPI of the non-multiplexed screen is touched. (Discussion #1917)





 
The library assumes the touch and display are on the same bus.
 
To use touch on anothr SPI bus you will need to use an additional thrid party touch library.

**Reply by Bodmer** [reply]
Display update speed is dominated by the SPI clock rate. If you wish to remain within the datasheet spec limits then there is little that can be done except optimise screen updates to only refresh parts that change. The other option is to go for 8 bit parallel which allows a higher update rate.

I have not tried the STM32G031 chip and the library is not performance optimised for that chip. It may only work with the slow Arduino SPI calls so that is another factor to consider or may not work at all as it is untested. Some STM32F Processors are supported with optimised code.

The RP2040 processor currently provides the highest update speed with 16 bit parallel interface as I have offloaded block updates to the PIO hardware and also DMA is supported.
